File: problems/two-sum/variants/python/75.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for index, value in enumerate(nums):
            complement = target - value
            if complement in seen:
                pass  # TODO: restore implementation; 75% solution supplied.
            seen[value] = index
        pass  # TODO: restore implementation; 75% solution supplied.

File: problems/two-sum/variants/cpp/75.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (int index = 0; index < static_cast<int>(nums.size()); ++index) {
            const int complement = target - nums[index];
            const auto match = seen.find(complement);
            if (match != seen.end()) return {match->second, index};
            // Hint: A naive O(n^2) solution will have a nested for loop to find (i,j) where nums[i] + nums[j] == target. Can you think of a better solution?
        }
        return {};
    }
};
